### PR TITLE
Polyculture Use Case: Add Plant Relationships

### PR DESCRIPTION
**Pull request by Benjaminimal**
This should document the following use case from https://github.com/ElektraInitiative/PermaplanT/issues/1
> adding companions to database on different levels

## Basics

- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.
- [x] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is conforming to [our Documentation Guidelines](/doc/documentation.md)
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to our Coding Guidelines
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

### CODE REVIEW

File: doc/usecases/polyculture/add_plant_relationships.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- **Scope:** Plants Layer
+- **Level:** User Goal
+- **Actors:** App User
+- **Brief:** A new plant relationship is added to the system.
+- **Status:** Draft
+
+## Scenarios
+
+- **Precondition:**
+  - The user has the opened the UI to propose a new relationship.
```
Comment (badnames) [inline doc/usecases/polyculture/add_plant_relationships.md:14]:
```suggestion
  - The user has opened the UI to propose a new relationship.
```

File: doc/usecases/polyculture/add_plant_relationships.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+- **Main success scenario:**
+  - Details about the relationship are provided by the user.
+     - The two sides of the relationship can be
+       - Specific plants
+       - Any higher taxonomic level (e.g. family, subfamily)
```
Comment (badnames) [inline doc/usecases/polyculture/add_plant_relationships.md:19]: Which taxonomic levels are there?
Please disregard this question, if this is not within the scope of this PR.

**Review by badnames — COMMENTED** [review]
Two minor changes, but looks good overall.

**Review by badnames — APPROVED** [review]

File: doc/usecases/polyculture/add_plant_relationships.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+     - The type of the relationship can be
+       - Companion
+       - Antagonist
+     - A description providing information on the whys and hows.
+- **Alternative scenario:**
+  - The user creates a relationship which is only effective for one or more of her selected maps.  
+    In this case there is no review process.
```
Comment (markus2330) [inline doc/usecases/polyculture/add_plant_relationships.md:26]:
```suggestion
    In this case there is no review process.
  - Map-specific relationships take preference.
```

File: doc/usecases/polyculture/add_plant_relationships.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+- **Error scenario:**
+  - A relationship with identical sides already exits.  
+    The user is informed about this and can't create the relationship.
```
Comment (markus2330) [inline doc/usecases/polyculture/add_plant_relationships.md:29]: See above, map-specific overrides are okay.

**Review by markus2330 — CHANGES_REQUESTED** [review]
Great job! Thanks @badnames for the review!

File: doc/usecases/polyculture/add_plant_relationships.md (hunk 2 above)
Comment (markus2330) [reply doc/usecases/polyculture/add_plant_relationships.md:19]: @Benjaminimal please define "relationship", "companion" and "antagonist" in the glossary.

Comment (Benjaminimal) [reply doc/usecases/polyculture/add_plant_relationships.md:19]: "companion" and "antagonist" are already in there.
I'll add the other term.
https://github.com/ElektraInitiative/PermaplanT/blob/c1240880cfdd47a5667c48aa0401c0a2e404b6ad/doc/architecture/glossary.md#L21-L25

File: doc/usecases/polyculture/add_plant_relationships.md (hunk 4 above)
Comment (Benjaminimal) [reply doc/usecases/polyculture/add_plant_relationships.md:29]: Ok, but the following should be an error too, right?

The user tries to add a map-specific one where a map-specific one with identical sides already exists.

Comment (markus2330) [reply doc/usecases/polyculture/add_plant_relationships.md:33]: Yes, this is an error.

File: doc/usecases/polyculture/add_plant_relationships.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+## Scenarios
+
+- **Precondition:**
+  - The user has opened the UI to propose a new relationship.
+- **Main success scenario:**
+  - Details about the relationship are provided by the user.
+     - The two sides of the relationship can be
+       - Specific plants
+       - Any higher taxonomic rank
```
Comment (markus2330) [inline doc/usecases/polyculture/add_plant_relationships.md:20]:
```suggestion
       - Taxonomic ranks which are not specific plants
```

File: doc/usecases/polyculture/add_plant_relationships.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+     - The type of the relationship can be
+       - Companion
+       - Antagonist
+       - Neutral (needed to indicate certainty about neither companion nor antagonist)
```
Comment (markus2330) [inline doc/usecases/polyculture/add_plant_relationships.md:24]:
```suggestion
       - Neutral
```

**Review by markus2330 — APPROVED** [review]
Great job!

**Pull request merged**